docs/introduction/entities.md

```diff
@@ -356,15 +356,59 @@ class BrandData
 }
 ```
 
-### Data for multidomain or multilanguage field
+### Recommendations for data object properties
 
-As you can see in example above, multidomain or multilanguage field is an array.
+#### Scalars
 
-Multidomain field like `seoH1s` have to be indexed by `domainId` - integer ID of the given domain.
-Data factory has to prepare this array to contain keys for all domain IDs, because domain entities are created by these array items (even if their value is null).
+Scalars are the most typical properties in data objects.
+You usually don't have a default value for a scalar field, so your phpdoc will usually look like `string|null`, `int|null`, `float|null`.
 
-Multilanguage field like `description` have to be indexed by `locale` - string identifier of language (you can find them in [`domains.yml`](/project-base/app/config/domains.yml)).
-Data factory does not need to prepare this array, because the extension we use for translated entities can handle unprepared translations.
+If you need to transfer boolean data, we recommend using `bool` with a default value in the constructor (`true`/`false`) because `false` and `null` behaves similarly in php.
+
+#### Arrays
+
+If you need to transfer arrays, use phpdoc `string[]`, `int[]`, `float[]`, `bool[]` and initialize an empty array in the constructor.
+
+If you care about array keys, use a name of the key in the property name in form `propertyByKey`. Eg. [`TransportData::$pricesByCurrencyId`](/packages/framework/src/Model/Transport/TransportData.php).
+
+#### Unknown types
+
+We don't recommend to use `mixed` or `array` phpdoc as they aren't expressive.
+
+The only exception in the framework is `$pluginData` in `CategoryData` and `ProductData`. The phpdoc is `array` in this case because plugins can contain any type.
```

Comment:
what about `in the Framework`

Reply:
we use small `framework` in this whole document, so I would keep it as it is